luten-scheduling/src/instances/random.rs

```diff
@@ -0,0 +1,202 @@
+                    for _ in 0..no {
+                        loop {
+                            let slots: Vec<_> = if block {
+                                (0..available_blocks_per_day).map(|n| n * 4).collect()
+                            } else {
+                                (0..(available_blocks_per_day * 4)).collect()
+                            };
+                            let slot = Timeslot {
+                                day: *(rng.choose(&[Monday, Tuesday, Wednesday]).unwrap()),
+                                slot_of_day: *(rng.choose(&slots).unwrap()),
+                            };
+
+                            if !map.contains_key(&slot) {
+                                map.insert(slot, rating);
+                                if block {
+                                    for i in 1..4 {
+                                        let new_slot = Timeslot {
+                                            day: slot.day,
+                                            slot_of_day: slot.slot_of_day + i,
+                                        };
+                                        map.insert(new_slot, rating);
+                                    }
+                                }
+                                break;
+                            }
+                        }
+
+                    }
+                };
+
+
+                add(good_blocks, true, SlotRating::Good);
+                add(tolerable_blocks, true, SlotRating::Tolerable);
+                add(good_slots, false, SlotRating::Good);
+                add(tolerable_slots, false, SlotRating::Tolerable);
+            }
+
+            SlotAssignment {
+                ratings: map,
+            }
+        }
+
+        let total_available_slots = self.available_blocks_per_day * 4 * 3;
+        let slot_no = {
+            let slot_no = self.rated_slots.ind_sample(rng).round();
+            clamp(slot_no, 1.0, total_available_slots as f64) as u64
+        };
+
+        let good_slots = {
+            let good_slots = (self.good_slot_percentage.ind_sample(rng) * (slot_no as f64)).round();
+            clamp(good_slots, 1.0, total_available_slots as f64) as u64
+        };
+        let tolerable_slots = slot_no - good_slots;
+
+        let block_percentage_sample = {
+            let percentage = self.good_slot_percentage.ind_sample(rng);
+            clamp(percentage, 0.0, 1.0)
+        };
+        let good_blocks = ((good_slots as f64) * block_percentage_sample / 4.0).round() as u64;
+        let good_single_slots = good_slots - (good_blocks * 4);
+        let tolerable_blocks = ((tolerable_slots as f64) * block_percentage_sample / 4.0)
+            .round() as u64;
+        let tolerable_single_slots = tolerable_slots - (tolerable_blocks * 4);
+
+        fill_slots(rng, good_blocks, tolerable_blocks, good_single_slots,
+            tolerable_single_slots, self.available_blocks_per_day)
+    }
+}
+
+pub fn random_instance<T, U, V, W, X, Y>(
```

Comment:
If you have that many generic parameters, it's perfectly fine to give proper names (it's preferred actually). In that case, we use `CamelCase` for those.

Reply:
And this function could need some doc comment explaining the parameters as well as what it does.